tests/compiler/cli/test.ts

```diff
@@ -0,0 +1,47 @@
+// tslint:disable-next-line:no-implicit-dependencies
+import execa = require('execa');
+
+jest.setTimeout(20000);
+test('Compiler should respect out-dir setting and write paths relative to folder', async () => {
+  const result = await execa(
+    process.argv[0],
+    [
+      '../../../dist/compiler/cli.js',
+      '--project',
+      './tsconfig.json',
+      '--out-dir',
+      './gen',
+      '--std-out'
+    ],
+    { cwd: __dirname }
+  );
+
+  expect(result.stdout).toBe(`// tslint:disable
+import { TSDI } from 'tsdi';
+import {
+  Container as C1_Container,
+  Dependency as C3_Dependency,
+  Test as C2_Test
+} from '../container';
+export class ContainerImpl extends C1_Container {
```

Comment:
I understand what this is needed for, but my guess would be that one will afterwards at project's toplevel layer import the `ContainerImpl`s from the generated code and instanciate them.

If you'd have some kind of static registry in tsdi (`ContainerRegistry.implFor(Container)`) we could avoid to force the user into a renaming.

The container impl's would self-register themselfs at the registry in the generated code below (`ContainerRegistry.register(ContainerImpl)`).

Just a rough idea.

Reply:
> I understand what this is needed for, but my guess would be that one will afterwards at project's toplevel layer import the ContainerImpls from the generated code and instanciate them.

Thats intended use. One should instantiate the container by itself since there is no runtime registry of TSDI anymore. You have to do this instead of creating a TSDI instance (this is done inside of the container).
The registering and accessing via factory method is a nice idea though. But then users still need to import the 'container impl file' somewhere in their project.